pupa/importers/base.py

```diff
@@ -234,14 +242,45 @@ def import_item(self, data):
 
         # obj existed, check if we need to do an update
         if obj:
+            _matched_obj_data = copy.deepcopy(obj.__dict__)
             if obj.id in self.json_to_db_id.values():
-                raise DuplicateItemError(data, obj)
+                if data.get('source_identified', False):
+                    obj_sources = set([(s.url, s.note) for s in obj.sources.all()])
+                    _matched_obj_data['sources'] = copy.deepcopy(obj_sources)
+                    possible_dupes = [k for k, v in self.json_to_db_id.items()
+                                      if v == obj.id]
+                    for pd in possible_dupes:
+                        pd_sources = self.json_to_sources[pd]
+                        if len(pd_sources & obj_sources) > 0:
+                            raise DuplicateItemError(data, obj)
+                else:
+                    raise DuplicateItemError(data, obj)
+
             # check base object for changes
             for key, value in data.items():
-                if getattr(obj, key) != value:
-                    setattr(obj, key, value)
-                    what = 'update'
+                obj_value = getattr(obj, key)
+                if key == "extras":
```

Comment:
AFAIK nothing has changed on master wrt `extras`, so this shouldn't present any problems.

what would need to change if we move to the JSONB type:
  - no more `json.loads` in line 263
  - `combine_dicts` wouldn't need to check the types on each